### ISSUE TITLE
Codeunit 12 "Gen. Jnl.-Post Line" UpdateCalcInterest Poor Performance

### ISSUE THREAD

**Issue by BjEic** [microsoft/ALAppExtensions#18607]
Hi,
I have a company with > 5 Mio open Customer Ledger Entries (Saas Environment) and I have to apply these entries. In procedure UpdateCalcInterest(var CVLedgEntryBuf: Record "CV Ledger Entry Buffer") ist this filter  CustLedgEntry.SetRange("Closed by Entry No.", "Entry No.");
The post of one application for one invoice with a payment will last more than 90 seconds.

Best reagrds
BjornE

**Comment by AndersLarsenMicrosoft** [comment]
Is this the code you are referring to?
    local procedure UpdateCalcInterest(var CVLedgEntryBuf: Record "CV Ledger Entry Buffer")
    var
        CustLedgEntry: Record "Cust. Ledger Entry";
        CVLedgEntryBuf2: Record "CV Ledger Entry Buffer";
    begin
        with CVLedgEntryBuf do begin
            if CustLedgEntry.Get("Closed by Entry No.") then begin
                CVLedgEntryBuf2.TransferFields(CustLedgEntry);
                UpdateCalcInterest(CVLedgEntryBuf, CVLedgEntryBuf2);
            end;
            CustLedgEntry.SetCurrentKey("Closed by Entry No.");
            CustLedgEntry.SetRange("Closed by Entry No.", "Entry No.");
            OnUpdateCalcInterestOnAfterCustLedgEntrySetFilters(CustLedgEntry, CVLedgEntryBuf);
            if CustLedgEntry.FindSet() then
                repeat
                    CVLedgEntryBuf2.TransferFields(CustLedgEntry);
                    UpdateCalcInterest(CVLedgEntryBuf, CVLedgEntryBuf2);
                until CustLedgEntry.Next() = 0;
        end;
    end; 
    
    It does a get by PK
    
    and later this
                CustLedgEntry.SetCurrentKey("Closed by Entry No.");
            CustLedgEntry.SetRange("Closed by Entry No.", "Entry No.");
            
     it looks ok at first glance

**Comment by AndersLarsenMicrosoft** [comment]
We could 
a) add a new key Customer No., Closed By Entry No
b) offer an event to skip updating "Calculate Interest" (used for Finance Charge Memo)
c) offer an event to set any filter
but the key question is what is the business reason for having so many open CLE and have many close entries to they have. Do they have a high amount of customers or?

**Comment by BjEic** [comment]
Hi Anders,
I think an event to skip Calculate Interest is a good solution. The Finance Charge feature is not used. Maybe for other customer is #a interesting

The reason why are there some manxy open CLE ist that we have to post about 25000 sales invoices per day. Every Invoice If business grows as in the past months we have to post more than 100000 invoices per day. We did some tests in the sandbox with up to 200000 invoices per day to show that the SaaS Enviroment has enough performance.

**Comment by AndersLarsenMicrosoft** [comment]
I will go for a combination of b and c hopefully it will make it for 20.3

**Comment by AndersLarsenMicrosoft** [comment]
You can use this event OnApplyCustLedgEntryOnAfterCalcShouldUpdateCalcInterestFromOldBuf to skip the code - we will leave it with this for now.

**Comment by AndersLarsenMicrosoft** [comment]
As state above there already is ab event to unblock you. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.
